### PR TITLE
Support for electron global keyboard shortcuts

### PR DESCRIPTION
**Pull request by baumhoto**
this one is based on #128 and additionally adds global shortcut keys to electron-app, so copying different attributes to clipboard (user, password, url) doesn't require to switch back to app once the entry is selected.

Global shortcut keys added for copying attributes to clipboard: password(c), user(u), url(u)
OSX: Ctrl+Alt+(c/b/u) 
win/linux: Shift+Alt+(c/b/u)

Don't know if that is a feature you want to support?

### CODE REVIEW

**Comment by antelle** [comment]
I've closed the previous PR as this includes it.
So, to merge it, please fix the following:
- join logic from [copyUserKeyPress](https://github.com/antelle/keeweb/blob/develop/app/scripts/views/details/details-view.js#L299) and copyUrlKeyPress to common method
- add shortcuts description to [settings/shortcuts](https://github.com/antelle/keeweb/blob/develop/app/templates/settings/settings-shortcuts.hbs#L10) page with corresponding [locale key](https://github.com/antelle/keeweb/blob/develop/app/scripts/util/locale.js#L291) - system-wide shortcuts should be the last
- style and spacing comments, I'll add them to the code

File: electron/app.js [hunk 1]
```diff
@@ -44,6 +46,17 @@ app.on('window-all-closed', function() {
     }
 });
 app.on('ready', function() {
+    if (process.platform === 'darwin') {
+        createGlobalShortcut('Ctrl+Alt+B', 'copy-user');
+        createGlobalShortcut('Ctrl+Alt+C', 'copy-password');
+        createGlobalShortcut('Ctrl+Alt+U', 'copy-url');
+    }
```
Comment (antelle) [inline electron/app.js]: `} else {`

File: electron/app.js [hunk 2]
```diff
@@ -58,6 +71,9 @@ app.on('activate', function() {
         }
     }
 });
+app.on('will-quit', function() {
+  globalShortcut.unregisterAll();
```
Comment (antelle) [inline electron/app.js]: 4 spaces

File: electron/app.js [hunk 3]
```diff
@@ -235,3 +251,14 @@ function notifyOpenFile() {
         openFile = null;
     }
 }
+
+function createGlobalShortcut(shortcut, backboneEventname)
```
Comment (antelle) [inline electron/app.js]: ) {

File: electron/app.js [hunk 4]
```diff
@@ -235,3 +251,14 @@ function notifyOpenFile() {
...
+{
```
Comment (antelle) [inline electron/app.js]: 4 spaces

File: electron/app.js [hunk 5]
```diff
@@ -235,3 +251,14 @@ function notifyOpenFile() {
...
+    var ret = globalShortcut.register(shortcut, function() {
+        emitBackboneEvent(backboneEventname);     
+    });
+
+  if (!ret) {
+    console.log('registration failed');
```
Comment (antelle) [inline electron/app.js]: just remove it please

File: electron/app.js [hunk 6]
```diff
@@ -44,6 +46,17 @@ app.on('window-all-closed', function() {
...
+    if (process.platform === 'darwin') {
+        createGlobalShortcut('Ctrl+Alt+B', 'copy-user');
+        createGlobalShortcut('Ctrl+Alt+C', 'copy-password');
+        createGlobalShortcut('Ctrl+Alt+U', 'copy-url');
+    }
+    else
+    {
+        createGlobalShortcut('Shift+Alt+B', 'copy-user');
+        createGlobalShortcut('Shift+Alt+C', 'copy-password');
+        createGlobalShortcut('Shift+Alt+U', 'copy-url');
```
Comment (antelle) [inline electron/app.js]: please make common event `copy-field` parametrized with the field itself

Comment (baumhoto) [inline electron/app.js]: don't understand what i need to do here, so if you do it i'm fine with that:)

**Comment by baumhoto** [comment]
i've updated the commit and fixed most of the issues.
- i actually refactored the 3 copyKeyPress methods into 1. hope your fine with that
- fixed code formatting issues etc.
- i modified the shortcuts-settings page, but there are still some issues:
  - it seems that this shortcut.css style does not support control key (at least not on the mac), or it does not support 2 modifier keys? it only display the Alt (Option) Key
  - in general this global modifier keys should only be displayed if user runs the electron app, but not if he uses the website, but don't know how i could do this in hbs file
 - currently  mac uses different global shortcut keys (control instead of shift), thus depending on the underlying platform the global shortcut keys differ. reason for using different modifier keys is that I'm used the mac combination (Ctrl+Alt) which does not work on windows. i have no problem to change it to Shift+Alt on mac for this pull request

**Comment by antelle** [comment]
Okay, I'll fix this myself and merge this manually tonight. Actually, currently there's a bug with unsubscribe [here](https://github.com/antelle/keeweb/pull/129/files#diff-dc353f1b741bbe10d926b0e6eeee948cR72) which is not performed, as subscribe is made with an anonymous function and unsubscribe with named - I'll fix it as well.

**Comment by baumhoto** [comment]
thanks, yes you're absolutely right i missed those calls in unsubscribe

**Comment by antelle** [comment]
Merged it, thank you! Build with your changes deployed to https://beta.keeweb.info/

**Pull request merged**